### ISSUE TITLE
[ISSUE] databricks_azure_adls_gen2_mount fails during PLAN phase because cluster does not exist

### ISSUE THREAD

**Issue by Satyricon** [databrickslabs/terraform-provider-databricks#803]
Hello,

I am trying to understand why are we getting the following error when running our databricks configuration Terraform project that is using `databricks_azure_adls_gen2_mount` resource.

```
│ Error: 0825-053542-fuels59 is not able to transition from TERMINATED to RUNNING: Self-bootstrap failure during launch. Please try again later and contact Databricks if the problem persists. databricks_error_message: Self-bootstrap failure during launch. Please try again later a..., Termination info: code: SELF_BOOTSTRAP_FAILURE, type: , parameters: map[databricks_error_message:Self-bootstrap failure during launch. Please try again later and contact Databricks if the problem persists. VM self bootstrap failed [Command_SetupWorker]]. Please see https://docs.databricks.com/dev-tools/api/latest/clusters.html#clusterclusterstate for more details
│ 
│   with databricks_azure_adls_gen2_mount.p360_executables,
│   on mount.tf line 21, in resource "databricks_azure_adls_gen2_mount" "p360_executables":
│   21: resource "databricks_azure_adls_gen2_mount" "p360_executables" {
│ 
╵
##[error]Bash exited with code '1'.
```

We create the mount using `cluster_id = null` option, so relying on provider to create the cluster for us.
Looking to Workspace I can't find `terraform-mount` cluster anymore.

My only guess is this: if you don't run your pipeline for over 30 days, databricks deletes the inactive (unpinned) cluster and provider is not able to find it anymore, so will not be able to perform mount check. Does this correspond to teh reality?

We are using databricks provider version `0.3.4`.

**Comment by nfx** [comment]
@Satyricon I think I've fixed this issue in some of the later releases. Please update to 0.3.7 and tell if the issue persists.

Otherwise file a support ticket with error messages - looks like a platform cluster manager issue

**Comment by Satyricon** [comment]
Thanks @nfx! I will test with latest version

**Comment by Satyricon** [comment]
It didn't work. Still the similar error `0825-100237-bunt436 is not able to transition from TERMINATED to RUNNING`.
@nfx - can you explain the logic how should it work? If I delete  `terraform-mount` cluster, should trhe provider be able to recover from it and recreate it?

**Comment by nfx** [comment]
@Satyricon this is not the provider issue - `Self-bootstrap failure during launch. Please try again later and contact Databricks if the problem persists. databricks_error_message: Self-bootstrap failure during launch. Please try again later a..., Termination info: code: SELF_BOOTSTRAP_FAILURE` - please create a support ticket with this message. you probably cannot create other clusters in that workspace as well.

It was fixed in 0.3.7 indeed - https://github.com/databrickslabs/terraform-provider-databricks/commit/4528b204479ec58b938fb74791ee79cfb6786fca#diff-0cd72fd40104d2cc80be375080fcdbb08ce5deb79eec724f200b64236fe36137

**Comment by nfx** [comment]
Duplicate of #637
